### PR TITLE
Add tracking large nft asset json strings, generating errors for crashlytics #4565

### LINKED ISSUE #4565: `RLMException, reason: String too big`
@hboon do you have any ideas of fixing this issue?

https://console.firebase.google.com/u/0/project/alphawallet-ios-prod/crashlytics/app/ios:com.stormbird.alphawallet/issues/087983306e9dd43e58732fcfac54a7b9?time=last-twenty-four-hours&sessionEventKey=90639cde2eb6426081d6f36049c5d468_1674795446825436627

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4565

### CODE REVIEW

**Comment by hboon** [comment]
Haven’t check it yet, but if you are loading the JSON, can you log the lengthy instead of the content? It’s huge :)

File: AlphaWallet/Core/Helpers/FirebaseReportService.swift [hunk 1]
```diff
@@ -55,4 +55,67 @@ extension Crashlytics {
...
     }
+
+    /// Logs large nft asset jsons to Realm, that cause crashes
+    /// - actions - update nft token balance operations
+    /// - fileSizeThreshold - max json string size, that will trigger logging
+    @discardableResult func logLargeNftJsonFiles(for actions: [AddOrUpdateTokenAction], fileSizeThreshold: Double = 10) -> Bool {
+        func logNftJsonFileIfNeeded(address: AlphaWallet.Address, server: RPCServer, balance: NonFungibleBalance) -> Bool {
+            switch balance {
+            case .assets(let rawAssets):
+                let rawAssets = rawAssets.compactMap { asset in asset.json.data(using: .utf8).flatMap { return (data: $0, value: asset) } }
```
Comment (hboon) [inline AlphaWallet/Core/Helpers/FirebaseReportService.swift:66]: We can avoid unnecessary computation (String -> Data) be it big or small.

Maybe just `rawAssets.map { $0.json.count }`?

And change the threshold, if necessary to compensate?

**Review by hboon — CHANGES_REQUESTED** [review]
Can you make `LogLargeNftJsonFilesTests.swift` smaller? A 2MB file is hard on git :) It can be just a tiny string as long as we can trigger the test somehow?

**Comment by oa-s** [comment]
> Can you make LogLargeNftJsonFilesTests.swift smaller? A 2MB file is hard on git :) It can be just a tiny string as long as we can trigger the test somehow?

yes, sure.

**Pull request merged**